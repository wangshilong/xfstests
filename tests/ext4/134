#! /bin/bash
# FS QA Test No. 134
#
# test to reproduce PV951636:
#   project quotas not updated if a file is mv'd into that directory
#
#-----------------------------------------------------------------------
# Copyright (c) 2006 Silicon Graphics, Inc.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/quota

_cleanup()
{
	cd /
	umount $SCRATCH_MNT 2>/dev/null
	rm -f $tmp.*
}
trap "_cleanup; exit \$status" 0 1 2 3 15

filter_ext4_quota()
{
	grep $SCRATCH_DEV 2>&1 | tr -s '[:space:]' | _filter_scratch
}

# real QA test starts here
_supported_fs ext4
_require_test

dir=$SCRATCH_MNT/project

cp /dev/null $seqres.full
chmod a+rwx $seqres.full	# arbitrary users will write here

umount $SCRATCH_DEV 2>/dev/null
_scratch_mkfs_ext4 -O quota,project >> $seqres.full 2>&1
_scratch_mount

mkdir $dir
src/setproject -v 1 $dir
chattr +P $dir

setquota -P 1 100M 100M 0 0 $SCRATCH_MNT
quota -v -P 1 | filter_ext4_quota
touch $dir/1
touch $dir/2
cp $dir/2 $dir/3

quota -v -P 1 | filter_ext4_quota

xfs_mkfile 1M $TEST_DIR/6

#try cp to dir
cp $TEST_DIR/6 $dir/6
quota -v -P 1 | filter_ext4_quota

#try mv to dir
mv $TEST_DIR/6 $dir/7
quota -v -P 1 | filter_ext4_quota

# success, all done
status=0
exit
