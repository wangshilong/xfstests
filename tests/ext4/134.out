QA output created by 134
 SCRATCH_DEV 4 102400 102400 1 0 0 
 SCRATCH_DEV 4 102400 102400 4 0 0 
 SCRATCH_DEV 1028 102400 102400 5 0 0 
 SCRATCH_DEV 2052 102400 102400 6 0 0 
